libtransmission/torrent.h

```diff
@@ -932,6 +1000,38 @@ struct tr_torrent final : public tr_completion::torrent_view
     bool start_when_stable = false;
 
 private:
+    [[nodiscard]] constexpr std::optional<uint16_t> effective_idle_limit_minutes() const noexcept
+    {
+        auto const mode = idle_limit_mode();
+
+        if (mode == TR_IDLELIMIT_SINGLE)
+        {
+            return idle_limit_minutes();
+        }
+
+        if (mode == TR_IDLELIMIT_GLOBAL && session->isIdleLimited())
+        {
+            return session->idleLimitMinutes();
+        }
+
+        return {};
+    }
+
+    [[nodiscard]] constexpr std::optional<size_t> idle_seconds(time_t now) const noexcept
+    {
+        auto const activity = this->activity();
+
+        if (activity == TR_STATUS_DOWNLOAD || activity == TR_STATUS_SEED)
+        {
+            if (auto const latest = std::max(startDate, activityDate); latest != 0 && now >= latest)
```

Comment:
Could be a good idea to assert `now >= latest` here. I can't think of any use case where `now` isn't `tr_time()` at the moment.

Reply:
good call; added in 9cfa102764f2366368485c5e5e087112660bd395